### DISCUSSION TITLE
Unable to Find API to Fetch All Open Windows in an Application

### DISCUSSION THREAD

**Discussion by lindexi** [unoplatform/uno#17284] [Q&A] [answered]
Hello,

I am trying to find a way to fetch all the open windows within an application. However, I have not been able to find any API that could help me achieve this. I also did not find any API definition in WinUI3 that could implement this functionality. 

Is there an API in UNO that implements this functionality and can be used? 

Any help would be greatly appreciated. 

Thank you.

Reference: https://github.com/microsoft/microsoft-ui-xaml/issues/4525

**Comment by MartinZikmund** [comment, marked as answer]
Yes unfortunately there is no such api in WinUI yet (although the Application.Windows property actually exists if you look into the sources!). We currently offer ApplicationHelper.Windows which you can use on Uno targets, but it does not work on Windows. The best way currently would be for you to keep track of windows yourself inside your app

**Reply by lindexi** [reply]
Thank you. And I will try the `ApplicationHelper.Windows` api.
